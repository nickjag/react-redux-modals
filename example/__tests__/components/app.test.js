import React from 'react';
import { Provider } from 'react-redux';
import { ShallowMock } from '../helpers/shallow_mock';
import ConnectedApp, { App } from '../../src/components/app';
import Button from '../../src/components/button';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('app component', () => {
  test('renders the component', () => {

    const initialState = {};
    const store = mockStore(initialState);
    const wrapper = shallow(ShallowMock(<App />, store));

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('fires first modal action', () => {

    const initialState = {};
    const store = mockStore(initialState);
    const wrapper = mount(<Provider store={store}>
      <ConnectedApp />
    </Provider>);

    wrapper.find(Button).first().simulate('click');
    expect(store.getActions()).toMatchSnapshot();
  });

});