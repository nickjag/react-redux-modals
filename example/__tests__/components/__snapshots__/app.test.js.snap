// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component fires first modal action 1`] = `
Array [
  Object {
    "payload": Object {
      "modalProps": undefined,
      "modalType": "MODAL_AGREE",
    },
    "type": "react-redux-modals/SHOW",
  },
]
`;

exports[`app component renders the component 1`] = `
<div
  className="app"
>
  <h1>
    react-redux-modals
  </h1>
  <article>
    <p>
      <Button
        onClick={[Function]}
      >
        Simple agreement
      </Button>
    </p>
    <p>
      <Button
        onClick={[Function]}
      >
        With fade transition
      </Button>
    </p>
    <p>
      <Button
        onClick={[Function]}
      >
        Multiple modals
      </Button>
    </p>
    <p>
      <Button
        onClick={[Function]}
      >
        No escaping allowed
      </Button>
    </p>
    <p>
      <Button
        onClick={[Function]}
      >
        Pass props between
      </Button>
    </p>
  </article>
  <a
    className="npm"
    href="https://www.npmjs.com/package/react-redux-modals"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      height="14"
      viewBox="0 0 18 7"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h18v6H9v1H5V6H0V0zm1 5h2V2h1v3h1V1H1v4zm5-4v5h2V5h2V1H6zm2 1h1v2H8V2zm3-1v4h2V2h1v3h1V2h1v3h1V1h-6z"
        fill="#CB3837"
      />
      <path
        d="M1 5h2V2h1v3h1V1H1zM6 1v5h2V5h2V1H6zm3 3H8V2h1v2zM11 1v4h2V2h1v3h1V2h1v3h1V1z"
        fill="#FFF"
      />
    </svg>
  </a>
</div>
`;
